### ISSUE TITLE
training is not robustness

### ISSUE THREAD

**Issue by asdfqwer2015** [HaydenFaulkner/pytorch.repmet#3]
Hi, I tried to train a resnet on pets dataset(split with seen 28 classes, unseen 3 dog classes and 6 cat classes) trainval set seen classes and test set is used for eval unseen classes and seen classes. The pretrained resnet can reach 60% unseen acc. And I trained 100 epochs(25 episodes per epoch, m=12, d=4) 1-shot in eval steps.

a. backbone and DML jointly trained
The seen val acc > 85%, and I trained three times, the val unseen acc is 71%(with kmeans init), 72%(with kmeans init), and 75%(without kmeans init). The val unseen acc decreased in about 60epochs and then start to increasing.

b. only DML trained
The seen val acc > 85% too, and the unseen val acc decreased from 60%(pretrained resnet) to 44%.

BTW, In unseen eval, I merged one shot support embed and seen reps as `all_reps` to calculate dists.

The training seems not very robustness. What do I missed?
